python/ray/serve/tests/test_handle.py

```diff
@@ -225,7 +227,17 @@ def exists(self):
     assert "Available methods: ['exists']" in exception_string
 
 
-def test_handle_across_loops(serve_instance):
+def _get_asyncio_loop_running_in_thread() -> asyncio.AbstractEventLoop:
+    loop = asyncio.new_event_loop()
+    threading.Thread(
+        daemon=True,
+        target=loop.run_forever,
+    ).start()
+    return loop
+
+
+@pytest.mark.asyncio
+async def test_handle_across_loops(serve_instance):
```

Comment:
As far as I can tell, this test had a latent race condition where the loop could be closed under some circumstances. I've fixed that by running the loops in a separate thread.